src/components/Login/Login.js

```diff
@@ -33,11 +34,14 @@ const Login = ({ intl, location }) => {
   const validate = (values) => {
     const errors = {};
 
-    if (!values[fieldNames.user]) {
-      errors[fieldNames.user] = _('validation_messages.error_required_field');
-    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values[fieldNames.user])) {
-      // TODO: review if this is our desired validation
-      errors[fieldNames.user] = _('validation_messages.error_invalid_email_address');
+    let emailMsgError = validateRequiredField(values[fieldNames.user]);
+    if (emailMsgError) {
+      errors[fieldNames.user] = _(emailMsgError);
+    } else {
+      emailMsgError = validateEmail(values[fieldNames.user]);
+      if (emailMsgError) {
+        errors[fieldNames.user] = _(emailMsgError);
+      }
```

Comment:
Mmm, in general if inside if are difficult to be read, what do you think about this:

```js
const emailMsgError = validateRequiredField(values[fieldNames.user]) || validateEmail(values[fieldNames.user]);
if (emailMsgError) {
  errors[fieldNames.user] = _(emailMsgError);
}
```

At the beginning, it could be difficult to be read, but at the second look (and the rest ones) it is not.

Reply:
yes, good change!